lib/auth/oauth_gems.rb

```diff
@@ -226,5 +226,10 @@ def authorize_params
       end
     end
   end
+
+  def callback_url
+    options[:redirect_uri] || (full_host + script_name + callback_path)
```

Comment:
It also seems `omniauth-oauth2` has some problems. Anyway, it's not relevant here. https://github.com/intridea/omniauth-oauth2/pull/100